sound/soc/sof/intel/hda-stream.c

```diff
@@ -312,6 +324,9 @@ int hda_dsp_stream_put(struct snd_sof_dev *sdev, int direction, int stream_tag)
 		return -ENODEV;
 	}
 
+	if (use_link_dma)
+		snd_hdac_ext_stream_release(link_stream, HDAC_EXT_STREAM_TYPE_LINK);
+
```

Comment:
And do we really need to remember to pass "use_link_dma" to put? Shouldn't this be recorded in get and we allocated a linked link dma, we release it in put?

Reply:
IOW, @ujfalusi is adding automatic re-coupling in https://github.com/thesofproject/linux/pull/5525 

Not sure how we book-keep the case where we'd have multiple link DMAs per one host DMA, but shouldn't we have some state to track which case this is so the put() can do the right thing?